src/TimeService.Business/Helpers/Workdays/WorkTimeCreater.cs

```diff
@@ -0,0 +1,122 @@
+        {
+            const string logMessage = "Cannot get projects users.";
+
+            try
+            {
+                var response = _rcProjectsUsers.GetResponse<IOperationResult<IGetProjectsUsersResponse>>(
+                    IGetProjectsUsersRequest.CreateObj()).Result;
+
+                if (response.Message.IsSuccess)
+                {
+                    return response.Message.Body.ProjectUsers;
+                }
+
+                _logger.LogWarning(logMessage);
+            }
+            catch(Exception exc)
+            {
+                _logger.LogError(exc, logMessage);
+            }
+
+            return null;
+        }
+
+        private bool Execute()
+        {
+            DateTime time = DateTime.UtcNow;
+
+            Dictionary<Guid, List<Guid>> projectsUsers = GetProjectsUsers();
+
+            if (projectsUsers == null)
+            {
+                _previousAttempt = DateTime.UtcNow;
+                return false;
+            }
+
+            foreach (var pair in projectsUsers)
+            {
+                foreach (var userId in pair.Value)
+                {
+                    _workTimeRepository.Create(
+                        new DbWorkTime
+                        {
+                            Id = Guid.NewGuid(),
+                            Day = 1,
+                            Month = time.Month,
+                            Year = time.Year,
+                            ProjectId = pair.Key,
+                            UserId = userId
+                        });
+                }
+            }
+
+            _previousAttempt = DateTime.UtcNow;
+            _lastSuccessfulAttempt = _previousAttempt;
+            return true;
+        }
+
+        public WorkTimeCreater(
+            IWorkTimeRepository workTimeRepository,
+            IRequestClient<IGetProjectsUsersRequest> rcProjectsUsers,
+            ILogger<WorkTimeCreater> logger)
+        {
+            _workTimeRepository = workTimeRepository;
+            _rcProjectsUsers = rcProjectsUsers;
+            _logger = logger;
+        }
+
+        public void Start(
+            int minutesToRestartAfterError,
+            DateTime lastSuccessfulAttempt)
+        {
+            _minutesToRestart = minutesToRestartAfterError;
+            _lastSuccessfulAttempt = lastSuccessfulAttempt;
+
+            Task.Run(() =>
+            {
+                while (true)
+                {
+                    if (_lastSuccessfulAttempt == default
+                    || DateTime.UtcNow.Month != _lastSuccessfulAttempt.Month)
+                    {
+                        Execute();
+                    }
+
+                    if (_lastSuccessfulAttempt != _previousAttempt)
+                    {
+                        Thread.Sleep(_minutesToRestart * 60000);
+                    }
+                    else
+                    {
+                        Thread.Sleep(3600000);
```

Comment:
Can we use events instead?) It look hard

Reply:
and how?

Reply:
Mb use that? https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.monthcalendar.datechanged?view=net-5.0

Reply:
this only for windows forms app